v0/destinations/snapchat_custom_audience/transform.js

```
@@ -0,0 +1,188 @@
+  if (!message.type) {
+    throw new ErrorBuilder()
+      .setMessage(
+        "[snapchat_custom_audience]::Message Type is not present. Aborting message."
+      )
+      .setStatus(400)
+      .build();
+  }
+  if (!message.properties) {
+    throw new ErrorBuilder()
+      .setMessage(
+        "[snapchat_custom_audience]::Message properties is not present. Aborting message."
+      )
+      .setStatus(400)
+      .build();
+  }
+  if (!message.properties.listData) {
+    throw new ErrorBuilder()
+      .setMessage(
+        "[snapchat_custom_audience]::listData is not present inside properties. Aborting message."
+      )
+      .setStatus(400)
+      .build();
+  }
+  if (message.type.toLowerCase() !== "audiencelist") {
+    throw new ErrorBuilder()
+      .setMessage(
+        `[snapchat_custom_audience]::Message Type ${message.type} not supported.`
+      )
+      .setStatus(400)
+      .build();
+  }
+  if (!message.properties.listData.add && !message.properties.listData.remove) {
+    throw new ErrorBuilder()
+      .setMessage(
+        "[snapchat_custom_audience]::Neither 'add' nor 'remove' property is present inside 'listData'. Aborting message."
+      )
+      .setStatus(400)
+      .build();
+  }
+};
+
+const responseBuilder = (metadata, message, { Config }, type) => {
+  const { schema, segmentId } = Config;
+  const payload = { users: [] };
+  const userPayload = { schema: [], data: [] };
+  let schemaType;
+  let hashedProperty;
+  // Normalizing and hashing ref: https://marketingapi.snapchat.com/docs/#normalizing-hashing
+
+  let userArray;
+  if (type === "add") {
+    userArray = message.properties.listData.add;
+  } else if (type === "remove") {
+    userArray = message.properties.listData.remove;
+  }
+
+  switch (schema) {
+    case "email":
+      schemaType = "EMAIL_SHA256";
+      userArray.forEach(element => {
+        hashedProperty = element?.email;
+        // what if email is not present in all cases?
```

- regarding the missing email case or any of the schema elements for that matter, we can omit that particular element and take rest of the others.
-here check what happens if none of the elements contain email / phone etc. Then we have to send empty payload, does snapchat accept that? If not we need to throw error, if all of the elements having missing schema field